### PR TITLE
bare ptrs not default serializable

### PR DESCRIPTION
**Pull request by robertjharrison**
make bare pointers not default serializable so that pts to world object go through correct path

### CODE REVIEW

**Comment by robertjharrison** [comment]
POD includes something like

struct A {
       something *ptr;
}

and nested variants thereof.   So the check for bare pointer is a functional but incomplete solution.   Overall I am happy with it.  The alternative seems to be eliminating POD entirely.

File: src/madness/world/type_traits.h [hunk 1]
```diff
@@ -61,7 +61,12 @@ namespace madness {
     /// This defines stuff that is serialiable by default rules ... basically anything contiguous
     template <typename T>
     struct is_serializable {
-        static const bool value = std::is_fundamental<T>::value || std::is_member_function_pointer<T>::value || std::is_function<T>::value  || std::is_function<typename std::remove_pointer<T>::type>::value || std::is_pod<T>::value;
+        static const bool value = \
+            std::is_fundamental<T>::value || \
+            std::is_member_function_pointer<T>::value || \
+            std::is_function<T>::value  || \
+            std::is_function<typename std::remove_pointer<T>::type>::value || \
+            (std::is_pod<T>::value && !std::is_pointer<T>::value);
```
Comment (evaleev) [inline src/madness/world/type_traits.h]: - per http://en.cppreference.com/w/cpp/types/is_pod std::is_pod will be deprecated
- I think this clause is to make trivially-copyable composite types built of serializable elements to be serializable. There is no way right now (without reflection) to properly check this. My suggestion is to say here `(std::is_class<T>::value || std::is_array<T>::value) && std::is_trivially_copyable<T>::value` to make intent as explicit as possible.

The shortcomings of `std::is_trivially_copyable` are discussed https://groups.google.com/a/isocpp.org/forum/#!topic/std-discussion/wphImiqfX7Y%5B1-25%5D. It's a loaded topic.

Comment (robertjharrison) [reply src/madness/world/type_traits.h]: OK.  U R suggesting the following ??

+        static const bool value = \
+            std::is_fundamental<T>::value || \
+            std::is_member_function_pointer<T>::value || \
+            std::is_function<T>::value  || \
+            std::is_function<typename std::remove_pointer<T>::type>::value || \
+            ((std::is_class<T>::value || std::is_array<T>::value) && std::is_trivially_copyable<T>::value);

Comment (evaleev) [reply src/madness/world/type_traits.h]:
```
static const bool value = \
std::is_arithmetic<T>::value || \
std::is_member_function_pointer<T>::value || \
std::is_function<T>::value  || \
std::is_function<typename std::remove_pointer<T>::type>::value || \
((std::is_class<T>::value || std::is_array<T>::value) && std::is_trivially_copyable<T>::value);
```

**Pull request merged**